 - Check for EVP_PKEY_get1_EC_KEY
 - Comment out inappropriate AC_CONFIG_SUBDIRS

--- a/configure.ac
+++ b/configure.ac
@@ -387,6 +387,12 @@
     AC_MSG_WARN([Failed to find libssl so OpenSSL will not be used. If it is installed you can try the --with-openssl=DIR argument]) ],
     [ -lcrypto ])
  fi
+
+ if test "$use_openssl" = "yes"; then
+  AC_CHECK_LIB(crypto, EVP_PKEY_get1_EC_KEY,
+   [AC_DEFINE(HAVE_EVP_PKEY_EC, 1, [Have EVP_PKEY_get1_EC_KEY])],
+   [AC_MSG_WARN([Disabling support for EC crypto])])
+ fi
 fi
 
 OPENSSL_LIBS=
@@ -516,7 +522,7 @@
 
 # If we still don't have it, we use our own
 if test $have_pcre != yes ; then
-  AC_CONFIG_SUBDIRS( libpcre )
+#  AC_CONFIG_SUBDIRS( libpcre )
   CPPFLAGS="-I\$(top_srcdir)/$LIBPCREDIR $CPPFLAGS"
   LIBPCRE_LIBS="$LIBPCREDIR/libpcre.a"
   PCRE_BUILD="build-pcre"
--- a/nse_ssl_cert.cc
+++ b/nse_ssl_cert.cc
@@ -434,7 +434,7 @@
 }
 
 int lua_push_ecdhparams(lua_State *L, EVP_PKEY *pubkey) {
-#ifdef EVP_PKEY_EC
+#ifdef HAVE_EVP_PKEY_EC
   EC_KEY *ec_key = EVP_PKEY_get1_EC_KEY(pubkey);
   const EC_GROUP *group = EC_KEY_get0_group(ec_key);
   int nid;
